pipeline/stage.go

```diff
@@ -105,3 +108,51 @@ func (s *StageSlice) Sanitize(driver string) *StageSlice {
 		return nil
 	}
 }
+
+// Empty returns true if the provided stage is empty.
+func (s *Stage) Empty() bool {
+	// return true if the container is nil
+	if s == nil {
+		return true
+	}
+
+	// return true if every stage field is empty
+	if len(s.Name) == 0 &&
+		len(s.Needs) == 0 &&
+		len(s.Steps) == 0 &&
+		len(s.Environment) == 0 {
+		return true
+	}
+
+	// return false if any of the ruletype is provided
+	return false
+}
+
+// MergeEnv takes a list of environment variables and attempts
+// to set them in the stage environment. If the environment
+// variable already exists in the stage, then this will
+// overwrite the existing environment variable.
+func (s *Stage) MergeEnv(environment map[string]string) error {
+	// check if the container is empty
```

Comment:
I think the comment here has a typo i.e. ~`container`~

```suggestion
	// check if the stage is empty
```